.github/workflows/uwp-autoconvert.yml

```diff
@@ -32,54 +35,62 @@ jobs:
         (Get-Content global.json).replace("latestFeature", "disable") | Set-Content global.json
 
     - name: Setup .NET SDK
-      uses: actions/setup-dotnet@v3
+      uses: actions/setup-dotnet@3447fd6a9f9e57506b15f895c5b76d3b197dc7c2 # v3
       with:
         global-json-file: global.json
 
     - name: Setup GitVersion
-      uses: gittools/actions/gitversion/setup@v0.9.15
+      uses: gittools/actions/gitversion/setup@31b3198b2de324472547fecec2c0f721b70c38e2 # v0.9.15
       with:
         versionSpec: '5.10.3'
 
     - name: GitVersion
       id: gitversion
-      uses: gittools/actions/gitversion/execute@v0.9.15
+      uses: gittools/actions/gitversion/execute@31b3198b2de324472547fecec2c0f721b70c38e2 # v0.9.15
       with:
         useConfigFile: true
         configFilePath: gitversion.yml
 
-    - name: setup git config
-      run: |
-        git config user.name "GitHub Actions Bot"
-        git config user.email "<>"
+    - name: setup git username
+      run: git config user.name "GitHub Actions Bot"
+
+    - name: Setup git email
+      run: git config user.email "<>"
 
     - name: Add msbuild to PATH
-      uses: microsoft/setup-msbuild@v1.1
+      uses: microsoft/setup-msbuild@34cfbaee7f672c76950673338facd8a73f637506 # v1.1
 
-    - run: |
-       echo "JAVA_HOME=$env:JAVA_HOME_11_X64" >> $env:GITHUB_ENV
-       echo "JavaSdkDirectory=$env:JAVA_HOME_11_X64" >> $env:GITHUB_ENV
-      name: Select JDK 11
+    - run: echo "JAVA_HOME=$env:JAVA_HOME_11_X64" >> $env:GITHUB_ENV
+      name: Select JDK 11 (Java home)
```

Comment:
@ceddy4395 Why do you consider multi-line steps a smell?

Reply:
Multi-line steps are considered smells because if something fails, it might not be clear what exactly has failed. Making it harder (then it already is usually) to debug failing pipelines. Curious to hear your opinion on this? 

Also, for this specific case I've split the run commands, but because you're altering environment variables, in hindsight it might also be possible to configure them using env construction instead like below: 
```
env:
  JAVA_HOME:...
  JavaSdkDirectory:...
```

Reply:
I see, that makes sense. The only disadvantage of doing so is related to duplicating step conditions, making the action more verbose if it repeats too much. Anyways, thanks for the changes!